src/availability/query_data.rs

```diff
@@ -429,3 +429,84 @@ pub(crate) fn payload_size<Types: NodeType>(payload: &Payload<Types>) -> u64 {
         Err(_) => 0,
     }
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "")]
+pub struct BlockSummaryQueryData<Types: NodeType> {
+    pub(crate) header: Header<Types>,
+    pub(crate) hash: BlockHash<Types>,
+    pub(crate) size: u64,
+    pub(crate) num_transactions: u64,
+    pub(crate) proposer_id: SignatureKey<Types>,
+}
+
+// Add some basic getters to the BlockSummaryQueryData type.
+impl<Types: NodeType> BlockSummaryQueryData<Types> {
+    pub fn header(&self) -> &Header<Types> {
+        &self.header
+    }
+
+    pub fn hash(&self) -> BlockHash<Types> {
+        self.hash
+    }
+
+    pub fn height(&self) -> u64 {
+        self.header.block_number()
+    }
+
+    pub fn size(&self) -> u64 {
+        self.size
+    }
+
+    pub fn num_transactions(&self) -> u64 {
+        self.num_transactions
+    }
+
+    pub fn proposer_id(&self) -> SignatureKey<Types> {
+        self.proposer_id.clone()
+    }
+}
+
+// Get the Proposer from the given header.  At the moment the proposer cannot
+// be determined from the header, however it is intended to in the future.
+// For now we will just generate one with dummy data.
+// TODO update this function to retrieve the proposer from the Header type once
+//      the Header type has been updated to include the proposer.
+pub fn get_proposer<Types: NodeType>(_header: &Header<Types>) -> SignatureKey<Types> {
+    use hotshot::types::SignatureKey;
+    let (key, _) = SignatureKey::generated_from_seed_indexed([0; 32], 0);
+    key
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "")]
+pub struct TransactionSummaryQueryData<Types: NodeType> {
+    pub(crate) hash: TransactionHash<Types>,
+    pub(crate) header: Header<Types>,
+    // We want a way to determine a summary for each rollup entry, without
+    // the data directly, but rather a summary of the data.
+    // For now, we'll roll with the `Payload` itself.
+    pub(crate) transaction: Transaction<Types>,
+}
+
+// Since BlockSummaryQueryData can be derived entirely from BlockQueryData, we
+// implement the From trait to allow for a seamless conversion using rust
+// contentions.
+impl<Types: NodeType> From<BlockQueryData<Types>> for BlockSummaryQueryData<Types>
+where
+    Payload<Types>: QueryablePayload,
+{
+    fn from(value: BlockQueryData<Types>) -> Self {
+        let payload = value.payload();
+        let size = payload_size::<Types>(&payload.clone());
+        let num_transactions = payload.len(value.metadata());
+
+        BlockSummaryQueryData {
+            header: value.header().clone(),
```

Comment:
Since we have ownership of `value` here, we could move out of header to avoid a clone:
```suggestion
            header: value.header,
```

Reply:
If we change this to just `value.header`, we end up with a partially borrowed reference for the next reference to `value.header` in the call to `get_proposer`.  The only way I have to resolve this at the moment is to add a `.clone()` to the `value.header` call as `BlockHeader` doesn't implement the `Copy` trait.

Unless you have a different suggestion?

Reply:
I see if we implement the suggestion / feedback for the `BlockQueryData<Types>`, this concern *may* be resolved.